test/helpers/ssltestlib.c

```diff
@@ -1191,8 +1191,14 @@ int create_ssl_objects(SSL_CTX *serverctx, SSL_CTX *clientctx, SSL **sssl,
 
     /* Up ref these as we are passing them to two SSL objects */
     SSL_set_bio(serverssl, c_to_s_bio, s_to_c_bio);
-    BIO_up_ref(s_to_c_bio);
-    BIO_up_ref(c_to_s_bio);
+
+    if (!BIO_up_ref(s_to_c_bio))
+        goto error;
```

Comment:
This is wrong and will lead to a double free. At this point we only have 1 ref to `s_to_c_bio` but it was passed to `serverssl`, and will get freed when `serverssl` gets freed. With this change we will free it once via `SSL_free(serverssl)` and once directly via `BIO_free`.